lib/validate.js

```diff
@@ -247,12 +247,7 @@ exports.generateUml = function generateUml(specPath, outputDir, options) {
 
 exports.updateEndResultOfSingleValidation = function updateEndResultOfSingleValidation(validator) {
   if (validator.specValidationResult.validityStatus) {
-    if (!(log.consoleLogLevel === 'json' || log.consoleLogLevel === 'off')) {
-      let consoleLevel = log.consoleLogLevel;
-      log.consoleLogLevel = 'info';
-      log.info('No Errors were found.');
-      log.consoleLogLevel = consoleLevel;
-    }
+    log.info('No Errors were found.');
```

Comment:
I don't know this code at all, but the fact that it was guarded by `log.consoleLogLevel === 'json'` makes me fear that automation may try to parse the output of this script as JSON. Or will this message not appear on stdout for some reasons I'm unaware of?

Reply:
The way I read this block, it was overwriting the desired consoleLogLevel in any case except for `json` and `off`. By removing the check, it should never overwrite the specified consoleLogLevel, including `json` and `off`. I may be dumb though, let me know if I've not mentally applied DeMorgans law :)

Reply:
Errr, I guess I'm making the assumption that whatever is converting stdout to JSON can handle a message it wouldn't have seen before. I'm also assuming that consoleLogLevel `off` knows to totally disregard things printed to `info`. I guess the safest way to handle this would be to just keep that check there, with the chance that it adds a little redundancy.

As @amarzavery knows this code base best, I'll let him weigh in.

Reply:
Yes, in cases where someone set the log level to error then they do not see anything on the command prompt when running oav manually. Folks said they did not know if everything was ok. Hence I did the above thing to explicitly say that "No errors were found." 👍 .

Reply:
Error is above info in the log level severity...

Reply:
Ah, thanks for the context, @amarzavery. With my PR in the specs repository to use the "min" mocha reporter, it should clear that up without needing to print the "No errors were found." message.

Reply:
But your usecase is genuine. While running this as a script you don't need that info statement.
We can have a variable named mode that can tell us how oav is being used. If the mode is cli then we can display the log.info(..) statement. If it is script then we can choose to not log that statement. 

If you are writing a script then the entry point is via the file indicated in "main" property of package.json. For cli it is the bin property https://github.com/Azure/oav/blob/master/package.json#L49-L52

Reply:
You can export a variable [here in lib/validate.js](https://github.com/Azure/oav/blob/master/lib/validate.js#L23) called `exports.isCliMode = true;` (We assume that folks would be running oav as a CLI).
Then in [index.js over here](https://github.com/Azure/oav/blob/master/index.js#L8) `validate.isCliMode = false;`.

```javadcript
if (validator.specValidationResult.validityStatus) {
    if (!(log.consoleLogLevel === 'json' || log.consoleLogLevel === 'off') && exports.isCliMode) {
      let consoleLevel = log.consoleLogLevel;
      log.consoleLogLevel = 'info';
      log.info('No Errors were found.');
      log.consoleLogLevel = consoleLevel;
    }
}
```

Reply:
Hmmm, that seems counter-intuitive to me. Maybe a simpler option would be to have the default consoleLogLevel set to 'info'? In my mind anyway, I'd rather have the tool behave consistently regardless of how it was invoked.

Reply:
I see. That is a valid point too.